### PR TITLE
update embedded platform build script

### PR DESCRIPTION
**Pull request by mamin506**
1. Add dsa_build.sh -- build DSA by dsa.tcl script and create dsa_build/ directory as workspace
2. Add peta_build.sh -- build petalinux project from a DSA file in <dsa_name>/ directory
3. Add pfm_build.sh -- build platform from a pfm.tcl script and create platform/ directory
4. rewrite petalinux.sh to build to use above scripts

This let us easier to build what we need instead of rebuild everything.

### CODE REVIEW

**Comment by xdavidz** [comment]
There scripts are very helpful to me. 3 small nits:
1) The dsa is being renamed to XSA or Shell or Platform. We can keep the current version unchanged and fix it once we see the new name from other dependencies.
2) petalinux-config --oldconfig seems obsoleted? should we use --silentconfig?
3) Do we have a README for the over all usage for those scripts? It seems that we should have a small instruction and recommend people start using the script. As long as we keep using those script, we can keep the script up-to-date and save time.

**Comment by mamin506** [comment]
Hi @xdavidz ,
1. Yes, we should change DSA to XSA in 2019.2.
2. Let's use --silentconfig after 2019.2. Since this option is new in Petalinux 2019.1, it will not work on petalinux 2018.x.
3. This is a good point. Let's create a internal confluence page as the next step.

**Comment by xdavidz** [comment]
> Hi @xdavidz ,
> 
> 1. Yes, we should change DSA to XSA in 2019.2.
> 2. Let's use --silentconfig after 2019.2. Since this option is new in Petalinux 2019.1, it will not work on petalinux 2018.x.
> 3. This is a good point. Let's create a internal confluence page as the next step.
Actually, what about a markdown text just beside those scripts? Something like:
https://github.com/Xilinx/XRT/blob/master/README.rst

https://github.com/Xilinx/XRT/tree/master/src/runtime_src/tools/scripts/README.rst or txt in case of being compiled? Could be very simple for scripts inside this directory.

File: src/runtime_src/tools/scripts/peta_build.sh [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+echo 'IMAGE_INSTALL_append = " zocl"'               >> $PETALINUX_IMAGE_BBAPPEND
+echo 'IMAGE_INSTALL_append = " opencl-headers-dev"' >> $PETALINUX_IMAGE_BBAPPEND
+echo 'IMAGE_INSTALL_append = " opencl-clhpp-dev"'   >> $PETALINUX_IMAGE_BBAPPEND
+
+echo " * Adding XRT Kernel Node to Device Tree"
+#echo "******************* Device Tree hook ********************"
+echo "cat ${RECIPE_DIR}/../../runtime_src/core/edge/fragments/xlnk_dts_fragment_mpsoc.dts >> recipes-bsp/device-tree/files/system-user.dtsi"
+cat ${RECIPE_DIR}/../../runtime_src/core/edge/fragments/xlnk_dts_fragment_mpsoc.dts >> recipes-bsp/device-tree/files/system-user.dtsi
+echo "cat ${ORIGINAL_DIR}/dsa_build/zcu102_fragment.dts >> recipes-bsp/device-tree/files/system-user.dtsi"
+cat ${ORIGINAL_DIR}/dsa_build/zcu102_fragment.dts >> recipes-bsp/device-tree/files/system-user.dtsi
```
Comment (chvamshi-xilinx) [inline src/runtime_src/tools/scripts/peta_build.sh]: Hi Min,

peta_build.sh contains following lines which are specific to zcu102.dts addtion. We should give some extra option to specify the file which contains all the dts paths.

example: --dts-file <file which contains all extra dts files>

Comment (mamin506) [reply src/runtime_src/tools/scripts/peta_build.sh]: Good eye! I will fix this in next commit.

**Comment by mamin506** [comment]
> > Hi @xdavidz ,
> > 
> > 1. Yes, we should change DSA to XSA in 2019.2.
> > 2. Let's use --silentconfig after 2019.2. Since this option is new in Petalinux 2019.1, it will not work on petalinux 2018.x.
> > 3. This is a good point. Let's create a internal confluence page as the next step.
> 
> Actually, what about a markdown text just beside those scripts? Something like:
> https://github.com/Xilinx/XRT/blob/master/README.rst
> 
> https://github.com/Xilinx/XRT/tree/master/src/runtime_src/tools/scripts/README.rst or txt in case of being compiled? Could be very simple for scripts inside this directory.


Good idea. I will do that in next commit.

File: src/runtime_src/tools/scripts/petalinux.sh [hunk 2]
```diff
@@ -49,152 +53,39 @@ ORIGINAL_DIR=$PWD
...
-echo "petalinux-build"
-petalinux-build
+# dsa_build.sh will create ${PLATFORM_NAME}/ directory for petalinux project
+${XRT_REPO_DIR}/src/runtime_src/tools/scripts/peta_build.sh ${ORIGINAL_DIR}/dsa_build/${PLATFORM_NAME}.dsa
 
-cd $ORIGINAL_DIR
 echo " * Copying PetaLinux boot files (from: $PWD)"
-cp ./${PETALINUX_NAME}/images/linux/image.ub ${ORIGINAL_DIR}/dsa_build/src/a53/xrt/image/image.ub
+cd $ORIGINAL_DIR
+cp ./${PLATFORM_NAME}/images/linux/image.ub 	${ORIGINAL_DIR}/dsa_build/src/a53/xrt/image/image.ub
```
Comment (Himanshu-xilinx) [inline src/runtime_src/tools/scripts/petalinux.sh]: is it ok to have src/a53/xrt path fixed? but this will make it specific for zynqmp

**Pull request merged**